chapters/arrays.tex

```diff
@@ -195,10 +195,14 @@ \section{Built-in Array Functions}\doublelabel{built-in-array-functions}
 \lstinline[mathescape=true]!size(C, $j$)! = $1$ for $n_{\mathrm{A}} + 1 \leq j \leq n$, \lstinline[mathescape=true]!C[$i_{1}$, $\ldots$, $i_{n_{\mathrm{A}}}$, 1, $\ldots$, 1]! =
 \lstinline[mathescape=true]!A[$i_{1}$, $\ldots$, $i_{n_{\mathrm{A}}}$]!\\ \hline
 \end{longtable}
+The argument $n$ must be literal or a constant that can be evaluated during translation because it determines
+the number of dimensions of the returned array.
 
 \begin{nonnormative}
-The function \lstinline!promote! cannot be used in Modelica, because the number of dimensions of the returned array cannot be determined at compile time
-if $n$ is a variable.  Below, \lstinline!promote! is only used for constant $n$.
+Below, \lstinline!promote! is only used for constant $n$.
+
+Varying $n$ for \lstinline!promote! complicates matrix handling as it
```

Comment:
Given what has already been said, I think this should at most be a non-normative explanation right after giving the rule.  If we should keep it, I think it should also relate directly to the rule:
> An $n$ that isn't a constant that can be evaluated during translation…

Reply:
Ok, added as well (without contraction).

Reply:
Seems fine, even if I fail to see what was a contradiction.

Reply:
Contraction! Or basically that I used "is not" instead of "isn't", as it seems more formal and thus better suited for a specification.

Reply:
Oups, that explains it.